### PR TITLE
Revert "Fix lang table multi-store data issue"

### PR DESCRIPTION
**Pull request by matks**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.6.x
| Description?  | Reverts PrestaShop/PrestaShop#11876 as it seems to have introduced major regression https://github.com/PrestaShop/PrestaShop/issues/15274
| Type?         | bug fix
| Category?     | CO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes https://github.com/PrestaShop/PrestaShop/issues/15274
| How to test?  | See ticket

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/15621)

### CODE REVIEW

**Review by PierreRambaud — DISMISSED** [review]
If we revert this change, we must add an upgrade script to insert `_lang` values

File: install-dev/upgrade/sql/1.7.6.2.sql [hunk 1]
```diff
@@ -0,0 +1,50 @@
+SET SESSION sql_mode = '';
+SET NAMES 'utf8';
+
+INSERT INTO `PREFIX_category_lang` SELECT
+`entity`.`id_category`, 1, `entity`.`id_lang`, `entity`.`name`, `entity`.`description`, `entity`.`link_rewrite`, `entity`.`meta_title`, `entity`.`meta_keywords`, `entity`.`meta_description`
+FROM `PREFIX_category_lang` entity
+LEFT JOIN `PREFIX_category_lang` entity2 ON `entity2`.`id_shop` = 1 AND `entity`.`id_category` = `entity2`.`id_category`
```
Comment (jolelievre) [inline install-dev/upgrade/sql/1.7.6.2.sql:7]: I don't get it, you force the `id_shop` to `1`, but select values from `entity2.id_shop = 2 AND null`
So no rows is inserted with id_shop values different from 1, and even in the select how can id_shop be both equal to 1 and null

Sure, I'm not a champion in SQL ^^ But I'm not the only one so maybe you could add a comment to explain what the query does?

File: src/Adapter/Category/QueryHandler/GetCategoryForEditingHandler.php [hunk 2]
```diff
@@ -66,7 +67,7 @@ public function handle(GetCategoryForEditing $query)
     {
         $category = new Category($query->getCategoryId()->getValue());
 
-        if (!$category->id) {
+        if (!$category->id || (!$category->isAssociatedToShop() && Shop::getContext() == Shop::CONTEXT_SHOP)) {
```
Comment (jolelievre) [inline src/Adapter/Category/QueryHandler/GetCategoryForEditingHandler.php:70]: The condition is a bit different, so maybe a dedicated exception should be thrown?
Like `InvalidShopContextException` ? or a very explicit `CategoryNotFoundInContextException`

**Review by jolelievre — COMMENTED** [review]
I don't understand everything in your query :D

File: src/Adapter/Category/QueryHandler/GetCategoryForEditingHandler.php (hunk 2 above)
Comment (PierreRambaud) [reply src/Adapter/Category/QueryHandler/GetCategoryForEditingHandler.php:70]: In fact it's the original behaviour for `CategoryNotFoundException` 😅

File: install-dev/upgrade/sql/1.7.6.2.sql (hunk 1 above)
Comment (PierreRambaud) [reply install-dev/upgrade/sql/1.7.6.2.sql:7]: I'm looking for rows that haven't data with id_shop = 1.
The `LEFT JOIN` combined with a `WHERE NULL` check for nonexistent rows.
This means `Give me id_category that have id_shop != 1 and don't have a row with the same id_category and id_shop 1`.

This kind of SQL request are often used for finding duplicates.

Comment (jolelievre) [reply install-dev/upgrade/sql/1.7.6.2.sql:7]: Alright I finally understood it ^^ Thanks for learning a new SQL technique

File: src/Adapter/Category/QueryHandler/GetCategoryForEditingHandler.php (hunk 2 above)
Comment (jolelievre) [reply src/Adapter/Category/QueryHandler/GetCategoryForEditingHandler.php:70]: Ok, I won't question this logic then 😛

**Review by Progi1984 — APPROVED** [review]

**Comment by marionf** [comment]
@matks should we reopen the issue fixed by #11876 ?

**Comment by PierreRambaud** [comment]
@marionf I don't think because it was the legacy behaviour and afaik it's mandatory to have all language saved in database :/

**Review by jolelievre — APPROVED** [review]
Thanks @PierreRambaud

**Comment by PierreRambaud** [comment]
Thanks @matks

**Comment by matks** [comment]
> @marionf I don't think because it was the legacy behaviour and afaik it's mandatory to have all language saved in database :/

So you think that https://github.com/PrestaShop/PrestaShop/pull/11876 is harmless and actually expected ?

**Comment by PierreRambaud** [comment]
@matks After reading how the ObjectModel works with language, yes it was the expected behavior..

**Pull request merged**